components/raftstore/src/store/peer.rs

```diff
@@ -939,6 +939,15 @@ where
     /// The index of last compacted raft log. It is used for the next compact
     /// log task.
     pub last_compacted_idx: u64,
+    /// Record the time of the last raft log compact, the witness should query
+    /// the leader periodically whether `voter_replicated_index` is updated
+    /// if CompactLog admin command isn't triggered for a while.
+    pub last_compacted_time: Instant,
+    /// When the peer is witness, and there is any voter lagging behind, the
+    /// log truncation of the witness shouldn't be triggered even if it's
+    /// force mode, and this item will be set to `true`, after all pending
```

Comment:
I'm wondering in case a TiKV is down for 30 minutes, how big is the raft log accumulated?

Reply:
I don't know yet, I need to actually test it.